Tests/FluentAssertions.Specs/Specialized/TaskAssertionSpecs.cs

```diff
@@ -185,6 +187,272 @@ public async Task When_task_completes_late_it_should_succeed()
+        {
+            // Arrange
+            Func<Task> action = null;
+
+            // Act
+            Func<Task> testAction = async () =>
+            {
+                using var _ = new AssertionScope();
+                await action.Should().ThrowWithinAsync<InvalidOperationException>(
+                    100.Milliseconds(), "because we want to test the failure {0}", "message");
+            };
+
+            // Assert
+            await testAction.Should().ThrowAsync<XunitException>()
+                .WithMessage("*because we want to test the failure message*found <null>*");
+        }
+
+        [Fact]
+        public async Task When_task_throws_fast_it_should_succeed()
+        {
+            // Arrange
+            var timer = new FakeClock();
+            var taskFactory = new TaskCompletionSource<bool>();
+
+            // Act
+            Func<Task> action = () =>
+            {
+                return taskFactory
+                    .Awaiting(t => (Task)t.Task)
+                    .Should(timer).ThrowWithinAsync<InvalidOperationException>(100.Milliseconds());
+            };
+            taskFactory.SetException(new InvalidOperationException("foo"));
+
+            // Assert
+            await action.Should().CompleteWithinAsync(1.Seconds());
+        }
+
+        [Fact]
+        public async Task When_task_throws_asynchronous_it_should_succeed()
+        {
+            // Arrange
+            var timer = new FakeClock();
+            var taskFactory = new TaskCompletionSource<bool>();
+
+            // Act
+            Func<Task> action = () =>
+            {
+                return Awaiting(() => (Task)taskFactory.Task)
+                    .Should(timer).ThrowWithinAsync<InvalidOperationException>(1.Seconds());
+            };
+            var _ = action.Invoke();
+            taskFactory.SetException(new InvalidOperationException("foo"));
+
+            // Assert
+            await action.Should().CompleteWithinAsync(1.Seconds());
+        }
+
+        [Fact]
+        public async Task When_task_not_completes_it_should_fail()
+        {
+            // Arrange
+            var timer = new FakeClock();
+            var taskFactory = new TaskCompletionSource<bool>();
+
+            // Act
+            Func<Task> action = () =>
+            {
+                return taskFactory
+                    .Awaiting(t => (Task)t.Task)
+                    .Should(timer).ThrowWithinAsync<InvalidOperationException>(
+                        100.Milliseconds(), "because we want to test the failure {0}", "message");
+            };
+            timer.Delay(200.Milliseconds());
+
+            // Assert
+            await action.Should().ThrowAsync<XunitException>().WithMessage(
+                "Expected a <System.InvalidOperationException> to be thrown within 100ms"
+                + " because we want to test the failure message,"
+                + " but no exception was thrown.");
```

Comment:
```suggestion
            // Arrange
            var timer = new FakeClock();
            var taskFactory = new TaskCompletionSource<bool>();

            // Act
            Func<Task> action = () =>
            {
                return taskFactory
                    .Awaiting(t => (Task)t.Task)
                    .Should(timer).ThrowWithinAsync<InvalidOperationException>(
                        100.Ticks(), "because we want to test the failure {0}", "message");
            };
            timer.Delay(101.Ticks());

            // Assert
            await action.Should().ThrowAsync<XunitException>().WithMessage(
                "Expected a <System.InvalidOperationException> to be thrown within 10.0µs"
                + " because we want to test the failure message,"
                + " but no exception was thrown.");
```